hexkit/protocols/eventsub.py

```diff
@@ -41,15 +42,17 @@ class MalformedPayloadError(RuntimeError):
         the type.
         """
 
-    topics_of_interest: list[str]
-    types_of_interest: list[str]
+    topics_of_interest: list[Ascii]
+    types_of_interest: list[Ascii]
 
-    def consume(self, *, payload: JsonObject, type_: str, topic: str) -> None:
+    @abstractmethod
+    def consume(self, *, payload: JsonObject, type_: Ascii, topic: Ascii) -> None:
         """Receive an event of interest and process it according to its type.
 
         Args:
             payload (JsonObject): The data/payload to send with the event.
             type_ (str): The type of the event.
             topic (str): Name of the topic to publish the event to.
         """
+        check_ascii([type_, topic])
```

Comment:
```suggestion
        check_ascii(type_, topic)
```
See suggestion for changing this in utils.